modules/relaychain/src/lib.rs

```diff
@@ -0,0 +1,156 @@
+// This file is part of Acala.
+
+// Copyright (C) 2020-2021 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # Module Relaychain
+//!
+//! This module is in charge of handling relaychain related utilities and business logic.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+use codec::{Decode, Encode, FullCodec};
+use sp_runtime::traits::StaticLookup;
+
+// mod mock;
+// mod tests;
+use frame_support::RuntimeDebug;
+use module_support::CallBuilder;
+use primitives::Balance;
+use sp_std::{boxed::Box, marker::PhantomData, vec::Vec};
+
+use frame_system::Config;
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum BalancesCall<T: Config> {
+	#[codec(index = 3)]
+	TransferKeepAlive(BalancesTransferKeepAliveCall<T>),
+}
+
+/// Relaychain balances.transfer_keep_alive call arguments
+#[derive(Clone, Encode, Decode, RuntimeDebug)]
+pub struct BalancesTransferKeepAliveCall<T: Config> {
+	/// dest account
+	pub dest: <T::Lookup as StaticLookup>::Source,
+	/// transfer amount
+	#[codec(compact)]
+	pub value: Balance,
+}
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum UtilityCall<RelaychainCall> {
+	#[codec(index = 1)]
+	AsDerivative(UtilityAsDerivativeCall<RelaychainCall>),
```

Comment:
Another way is doing like this to avoid having a struct for each parameter. But the current way is also fine by me.

```suggestion
	AsDerivative { index: u16, call: RelaychainCall },
```